pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -1507,24 +1556,42 @@ public CompletableFuture<Void> close(
         CompletableFuture<Void> closeFuture = new CompletableFuture<>();
 
         lock.writeLock().lock();
+        CompletableFuture<Void> inProgressTransferCloseTask = null;
         try {
             if (!disconnectClients) {
                 transferring = true;
```

Comment:
Now, `isClosingOrDeleting` acts as a barrier to distinguishing between two Topic closings, and using `transferring` here would complicate the logic. 

BTW, the variable `transferring` will not be reset if the close is failed. We can do this improvement at the next PR: "Merge the status of PersistentTopic".

Reply:
Good point on future failures.

Do you think we need to handle future failure cases in this PR(What's our retry logic here)? It appears that we will keep on returning failed futures instead of rerunning the close after this PR.

Reply:
> Do you think we need to handle future failure cases in this PR(What's our retry logic here)? It appears that we will keep on returning failed futures instead of rerunning the close after this PR.

After close fails, the state `isClosingOrDeleting` will be reset to `false`, then the next `topic.close` will set the variable `this.closeFutures` to a new object.